### PR TITLE
Add $set and $set_once support

### PR DESCRIPTION
**Pull request by paolodamico**
Adds `$set` and `$set_once` support to the library. Related to https://github.com/PostHog/posthog/issues/2684.

Further, makes things easier to test & run locally. Using `setup.py test` is deprecated, and so is `test_requires`. Thus, I changed the test file names and added support for pytest so we can still test things. I *didn't* change any CI configuration, which hopefully gives confidence that the new and old testing methods are on par.

If you're okay with this change, I can "upgrade" `ci.yaml` as well.

### CODE REVIEW

**Comment by paolodamico** [comment]
First time taking a stab at this library, so a thorough review would be appreciated.

**Comment by paolodamico** [comment]
Ping @macobo should we move this forward?

**Comment by yakkomajuri** [comment]
If we're doing `$set_once`, shouldn't we also do `$set`?

**Comment by Twixes** [comment]
Definitely agree with support `set` the same way.

File: posthog/test/test_client.py [hunk 1]
```diff
@@ -125,7 +185,7 @@ def test_basic_page(self):
 
     def test_basic_page_distinct_uuid(self):
         client = self.client
-        distinct_id = uuid4()
+        distinct_id = str(uuid4())
```
Comment (neilkakkar) [inline posthog/test/test_client.py:188]: Not sure how this worked before O.o

File: simulator.py [hunk 2]
```diff
@@ -71,6 +71,14 @@ def identify():
...
 
 
+def set_once():
+    posthog.set_once(
+        options.distinct_id,
+        properties=json_hash(options.traits),
+        context=json_hash(options.context),
+    )
+
+
```
Comment (mariusandra) [inline simulator.py:81]: Just a nit, but might be good to add "set" here for completion as well.

Comment (neilkakkar) [reply simulator.py:81]: Gotcha, makes sense

**Review by mariusandra — APPROVED** [review]

**Comment by mariusandra** [comment]
Can you please still fix the black error... and then feel free to merge it into master :).

**Pull request merged**